src/Compilers/Core/Portable/CommandLine/SarifErrorLogger.cs

```diff
@@ -103,7 +103,7 @@ protected void WriteResultProperties(Diagnostic diagnostic)
 
                     foreach (var pair in diagnostic.Properties.OrderBy(x => x.Key, StringComparer.Ordinal))
                     {
-                        _writer.Write(pair.Key, pair.Value);
+                        _writer.Write(pair.Key, pair.Value ?? string.Empty);
```

Comment:
Before my change (and before/ignoring annotations), a null value gets serialized as if it's the empty string.
My change here avoids a compiler warning due to nullable annotations.

Reply:
It looks like `JsonWriter.Write(string, string)` should be annotated as `Write(string, string?)` instead.

---
In reply to: [402565899](https://github.com/dotnet/roslyn/pull/43023#discussion_r402565899) [](ancestors = 402565899)

Reply:
> However I think it's misleading because JsonWriter.Write(string, string) does not preserve null values. It converts them to "". 

The annotation would indicate the method accepts a `null` value, it would not indicate how the value is used. The annotation on the `location` parameter in `Diagnostic.Create()` is similar. In that case, a `null` location is converted to `Location.None`.

---
In reply to: [402592535](https://github.com/dotnet/roslyn/pull/43023#discussion_r402592535) [](ancestors = 402592535)

Reply:
I've pushed a change to add that annotation.